main.go

```diff
@@ -32,9 +33,22 @@ func Handler(ctx context.Context, request events.APIGatewayProxyRequest) (string
 	}
 
 	if strings.HasPrefix(request.Body, "update=") {
-		err = sendMessage(request.Body)
-		if err != nil {
-			return "Error sending SNS message", err
+		sparqlQuery := sparql.NewQuery()
+		err = sparqlQuery.Parse(strings.NewReader(request.Body))
+
+		for _, part := range sparqlQuery.Parts {
+			if strings.ToUpper(part.Verb) == "INSERT" || strings.ToUpper(part.Verb) == "INSERT DATA" {
+				err = sendMessage("touch", strings.Join(uniqueSubjects(part.Graph), "\", \""), request.Body)
+				if err != nil {
+					return "Error sending SNS message", err
+				}
+			}
+			if strings.ToUpper(part.Verb) == "DELETE" || strings.ToUpper(part.Verb) == "DELETE DATA" {
+				err = sendMessage("delete", strings.Join(uniqueSubjects(part.Graph), "\", \""), request.Body)
```

Comment:
Hmm, help me work through this, especially whether `strings.Join` is going to do what we expect it to here.

Assuming `uniqueSubjects` returns a slice of strings that looks like this:

```go
[http://example.org/1 http://example.org/2 http://example.org/3]
```

Won't the `strings.Join` call here return:

```go
http://example.org/1", "http://example.org/2", "http://example.org/3
```

That is, where/how are we injecting the leading and trailing quotation marks?

Reply:
@mjgiarlo The leading and trailing quotes are injected in the sendMessage method down at line 77. I can move that up here if it makes more sense however.

Reply:
@aaron-collier I don't a strong preference where, but I do like the idea of keeping all the string-munging in the same place, and perhaps within `sendMessage` makes the most sense. How do you feel about the `strings.Join` call moving down there?